src/gui/pointcloud/qgspointcloudrendererpropertieswidget.cpp

```diff
@@ -0,0 +1,219 @@
+  connect( mMaxErrorUnitWidget, &QgsUnitSelectionWidget::changed, this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
+
+  syncToLayer( layer );
+}
+
+void QgsPointCloudRendererPropertiesWidget::setContext( const QgsSymbolWidgetContext &context )
+{
+  mMapCanvas = context.mapCanvas();
+  mMessageBar = context.messageBar();
+  if ( mActiveWidget )
+  {
+    mActiveWidget->setContext( context );
+  }
+}
+
+void QgsPointCloudRendererPropertiesWidget::syncToLayer( QgsMapLayer *layer )
+{
+  mLayer = qobject_cast< QgsPointCloudLayer * >( layer );
+
+  mBlockChangedSignal = true;
+  mOpacityWidget->setOpacity( mLayer->opacity() );
+  mBlendModeComboBox->setBlendMode( mLayer->blendMode() );
+
+  if ( mLayer->renderer() )
+  {
+    // set current renderer from layer
+    QString rendererName = mLayer->renderer()->type();
+
+    int rendererIdx = cboRenderers->findData( rendererName );
+    cboRenderers->setCurrentIndex( rendererIdx );
+
+    // no renderer found... this mustn't happen
+    Q_ASSERT( rendererIdx != -1 && "there must be a renderer!" );
+
+    mMaxErrorSpinBox->setValue( mLayer->renderer()->maximumScreenError() );
+    mMaxErrorUnitWidget->setUnit( mLayer->renderer()->maximumScreenErrorUnit() );
+  }
+
+  mBlockChangedSignal = false;
+}
+
+void QgsPointCloudRendererPropertiesWidget::apply()
+{
+  mLayer->setOpacity( mOpacityWidget->opacity() );
+  mLayer->setBlendMode( mBlendModeComboBox->blendMode() );
+
+  if ( mActiveWidget )
+    mLayer->setRenderer( mActiveWidget->renderer() );
+  else if ( !cboRenderers->currentData().toString().isEmpty() )
+  {
+    QDomElement elem;
+    mLayer->setRenderer( QgsApplication::pointCloudRendererRegistry()->rendererMetadata( cboRenderers->currentData().toString() )->createRenderer( elem, QgsReadWriteContext() ) );
+  }
+
+  mLayer->renderer()->setMaximumScreenError( mMaxErrorSpinBox->value() );
+  mLayer->renderer()->setMaximumScreenErrorUnit( mMaxErrorUnitWidget->unit() );
+}
+
+void QgsPointCloudRendererPropertiesWidget::rendererChanged()
+{
+  if ( cboRenderers->currentIndex() == -1 )
+  {
+    QgsDebugMsg( QStringLiteral( "No current item -- this should never happen!" ) );
+    return;
+  }
+
+  QString rendererName = cboRenderers->currentData().toString();
+
+  //Retrieve the previous renderer: from the old active widget if possible, otherwise from the layer
+  std::unique_ptr< QgsPointCloudRenderer > oldRenderer;
+  if ( mActiveWidget && mActiveWidget->renderer() )
+  {
+    oldRenderer.reset( mActiveWidget->renderer()->clone() );
+  }
+  else
+  {
+    oldRenderer.reset( mLayer->renderer()->clone() );
+  }
+
+  // get rid of old active widget (if any)
+  if ( mActiveWidget )
+  {
+    stackedWidget->removeWidget( mActiveWidget );
+
+    delete mActiveWidget;
+    mActiveWidget = nullptr;
+  }
+
+  QgsPointCloudRendererWidget *w = nullptr;
```

Comment:
i know FORTRAN is popular, but i prefer readable variable names :)

Reply:
Done

Reply:
sure np, that is why I selected "approved" as status of review :)